osquery/tables/system/linux/npm_packages.cpp

```diff
@@ -0,0 +1,110 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <string>
+
+#include <boost/filesystem.hpp>
+
+#include <osquery/filesystem.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+#include "osquery/core/conversions.h"
+#include "osquery/core/json.h"
+
+namespace fs = boost::filesystem;
+
+namespace osquery {
+namespace tables {
+
+const std::vector<std::string> kPackageKeys{"name", "version", "description"};
+
+const std::string kLinuxNodeModulesPath{"/usr/lib/"};
+
+void genPackageResults(const std::string& directory, QueryData& results) {
+  std::vector<std::string> packages;
+  resolveFilePattern(directory + "/node_modules/%/package.json", packages);
+
+  for (const auto& package_path : packages) {
+    std::string json;
+    if (!readFile(package_path, json).ok()) {
+      LOG(WARNING) << "Could not read package JSON: " << package_path;
+      continue;
+    }
+
+    auto doc = JSON::newObject();
+    if (!doc.fromString(json) || !doc.doc().IsObject()) {
+      LOG(WARNING) << "Could not parse JSON from: " << package_path;
+      continue;
+    }
+
+    Row r;
+    for (const auto& key : kPackageKeys) {
+      if (doc.doc().HasMember(key)) {
+        const auto& value = doc.doc()[key];
+        // npm has a schema for package.json, but it is loosely enforced. Some
+        // keys may be missing, so populate the columns we can
+        r[key] = (value.IsString()) ? value.GetString() : "";
+      }
+    }
+
+    // Manually get nested key (Author name)
+    if (doc.doc().HasMember("author")) {
+      const auto& author = doc.doc()["author"]["name"];
```

Comment:
Should you be checking the key "name" exists before accessing it?

Reply:
Yes, I've fixed it now